### PR TITLE
Feat[EGB-011]create students,  grades and record books controllers

### PR DESCRIPTION
**Pull request by stanislav-plisko**

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1,3 +1,4 @@
 web: env RUBY_DEBUG_OPEN=true bin/rails server
 css: yarn watch:css
 js: yarn build --watch
+css: yarn watch:css
```
Comment (Sergick12) [inline Procfile.dev]: better this remove

File: app/controllers/grades_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private
+  
+    def grade_params
+      params.require(:grade).permit(:record_book_id, :grade)
+    end
+  
+    def set_grade
+      @grade = Grade.find(params[:id])
+    end
+  end
```
Comment (Sergick12) [inline app/controllers/grades_controller.rb]: add empty line

File: app/controllers/record_books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private
+  
+    def record_book_params
+      params.require(:record_book).permit(:subject_id, :student_id, :teacher_id, :intermediate_attestation_id)
+    end
+  
+    def set_record_book
+      @record_book = RecordBook.find(params[:id])
+    end
+  end
```
Comment (Sergick12) [inline app/controllers/record_books_controller.rb]: It too

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    private
+  
+    def student_params
+      params.require(:student).permit(:user_id, :specialization_id, :group_id)
+    end
+  
+    def set_student
+      @student = Student.find(params[:id])
+    end
+  end
```
Comment (Sergick12) [inline app/controllers/students_controller.rb]: It too

File: app/views/grades/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<h1>Редактировать Оценку</h1>
+
+<%= simple_form_for @grade, html: { class: 'form-horizontal' } do |f| %>
+  <%= f.association :record_book, label: 'Ведомость', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+  <%= f.input :grade, label: 'Оценка', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+  <%= f.button :submit, 'Сохранить изменения', class: 'btn btn-primary mt-3' %>
+<% end %>
+
+<%= link_to 'Назад', grades_path, class: 'btn btn-secondary mt-3' %>
```
Comment (Sergick12) [inline app/views/grades/edit.html.erb]: It too

File: app/views/grades/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          <%= link_to 'Показать', grade, class: 'btn btn-default' %>
+          <%= link_to 'Редактировать', edit_grade_path(grade), class: 'btn btn-default' %>
+          <%= button_to 'Удалить', grade, method: :delete, data: { confirm: 'Вы уверены?' }, class: 'btn btn-default' %>
+        </td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<%= link_to 'Создать новую оценку', new_grade_path, class: 'btn btn-primary' %>
```
Comment (Sergick12) [inline app/views/grades/index.html.erb]: It too

File: app/views/grades/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<h1>Новая Оценка</h1>
+
+<%= simple_form_for @grade, html: { class: 'form-horizontal' } do |f| %>
+  <%= f.association :record_book, label: 'Ведомость', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+  <%= f.input :grade, label: 'Оценка', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+  <%= f.button :submit, 'Создать Оценку', class: 'btn btn-primary mt-3' %>
+<% end %>
+
+<%= link_to 'Назад', grades_path, class: 'btn btn-secondary mt-3' %>
```
Comment (Sergick12) [inline app/views/grades/new.html.erb]: It too

File: app/views/grades/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<h1>Оценка</h1>
+
+<p>Ведомость: <%= @grade.record_book.subject.name %> (<%= @grade.record_book.student.user.name %>) </p>
+<p>Оценка: <%= @grade.grade %></p>
+
+<%= link_to 'Редактировать', edit_grade_path(@grade), class: 'btn btn-default' %>
+<%= link_to 'Назад', grades_path, class: 'btn btn-secondary' %>
```
Comment (Sergick12) [inline app/views/grades/show.html.erb]: It too

File: app/views/record_books/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    <%= form.collection_select :teacher_id, User.with_role(:teacher), :id, :name, {}, { class: 'form-control' } %>
+  </div>
+  
+    <div class="mb-3">
+      <%= form.label :intermediate_attestation_id, "Аттестация" %>
+      <%= form.collection_select :intermediate_attestation_id, IntermediateAttestation.all, :id, :name, {}, { class: 'form-control' } %> 
+    </div>
+  
+    <%= form.submit 'Сохранить ведомость', class: 'btn btn-primary' %>
+  <% end %>
```
Comment (Sergick12) [inline app/views/record_books/_form.html.erb]: All files

File: app/views/record_books/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<h1>Новая ведомость</h1>
+
+<%= render 'form', record_book: @record_book %> 
+
+<%= link_to 'Назад', record_books_path, class: 'btn btn-light' %>
```
Comment (Sergick12) [inline app/views/record_books/new.html.erb]: Dont use Russian

File: app/views/students/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,32 @@
+<h1>Студенты</h1>
+
+<% if @students.present? %>
+  <table class="table table-striped">
+    <thead>
+      <tr>
+        <th>ФИО</th>
```
Comment (Sergick12) [inline app/views/students/index.html.erb]: Dont use rusian in all files

File: spec/controllers/students_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,51 @@
+#require 'rails_helper'
```
Comment (Sergick12) [inline spec/controllers/students_controller_spec.rb]: remove folder spec/controllers

**Review by Sergick12 — CHANGES_REQUESTED** [review]

File: app/views/grades/_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+<%= simple_form_for grade, html: { class: 'form-horizontal' } do |f| %>
+    <%= f.association :record_book, label: 'Record Book', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+    <%= f.input :grade, label: 'Grade', required: true, wrapper: :horizontal_form, input_html: { class: 'form-control' } %>
+    <%= f.button :submit, class: 'btn btn-primary mt-3' %>
+  <% end %>
+
```
Comment (Baron-Dmitry) [inline app/views/grades/_form.html.erb]: add empty line

File: app/models/grade.rb [hunk 14]
```diff
@@ -1,4 +1,8 @@
 class Grade < ApplicationRecord
   belongs_to :record_book
   validates :grade, presence: true
+
+  has_one :subject, through: :record_book
```
Comment (Baron-Dmitry) [inline app/models/grade.rb]: Why does this association exist

File: app/models/grade.rb
Comment (Baron-Dmitry) [inline app/models/grade.rb:1]: look at the database diagram, what rows and what connections should be

**Review by Baron-Dmitry — CHANGES_REQUESTED** [review]

**Review by Baron-Dmitry — DISMISSED** [review]

**Pull request merged**